packages/react-core/src/Authenticator/hooks/useAuthenticatorRoute/useAuthenticatorRoute.ts

```diff
@@ -0,0 +1,122 @@
+import { useMemo } from 'react';
+import { AuthenticatorRouteComponentName } from '../types';
+import { useAuthenticator } from '../useAuthenticator';
+
+import { UseAuthenticatorRoute, UseAuthenticatorRouteParams } from './types';
+import {
+  getRouteSelector,
+  resolveConfirmResetPasswordRoute,
+  resolveConfirmSignInRoute,
+  resolveConfirmSignUpRoute,
+  resolveDefault,
+  resolveConfirmVerifyUserRoute,
+  resolveForceNewPasswordRoute,
+  resolveResetPasswordRoute,
+  resolveSetupTOTPRoute,
+  resolveSignInRoute,
+  resolveSignUpRoute,
+  resolveVerifyUserRoute,
+} from './utils';
+
+export default function useAuthenticatorRoute<PlatformProps>(
+  params: UseAuthenticatorRouteParams<PlatformProps>
+): UseAuthenticatorRoute<PlatformProps, 'ConfirmResetPassword'>;
+export default function useAuthenticatorRoute<PlatformProps>(
+  params: UseAuthenticatorRouteParams<PlatformProps>
+): UseAuthenticatorRoute<PlatformProps, 'ConfirmSignIn'>;
+export default function useAuthenticatorRoute<PlatformProps>(
+  params: UseAuthenticatorRouteParams<PlatformProps>
+): UseAuthenticatorRoute<PlatformProps, 'ConfirmSignUp'>;
+export default function useAuthenticatorRoute<PlatformProps>(
+  params: UseAuthenticatorRouteParams<PlatformProps>
+): UseAuthenticatorRoute<PlatformProps, 'ConfirmVerifyUser'>;
+export default function useAuthenticatorRoute<PlatformProps>(
+  params: UseAuthenticatorRouteParams<PlatformProps>
+): UseAuthenticatorRoute<PlatformProps, 'ForceNewPassword'>;
+export default function useAuthenticatorRoute<PlatformProps>(
+  params: UseAuthenticatorRouteParams<PlatformProps>
+): UseAuthenticatorRoute<PlatformProps, 'ResetPassword'>;
+export default function useAuthenticatorRoute<PlatformProps>(
+  params: UseAuthenticatorRouteParams<PlatformProps>
+): UseAuthenticatorRoute<PlatformProps, 'SetupTOTP'>;
+export default function useAuthenticatorRoute<PlatformProps>(
+  params: UseAuthenticatorRouteParams<PlatformProps>
+): UseAuthenticatorRoute<PlatformProps, 'SignIn'>;
+export default function useAuthenticatorRoute<PlatformProps>(
+  params: UseAuthenticatorRouteParams<PlatformProps>
+): UseAuthenticatorRoute<PlatformProps, 'SignUp'>;
+export default function useAuthenticatorRoute<PlatformProps>(
+  params: UseAuthenticatorRouteParams<PlatformProps>
+): UseAuthenticatorRoute<PlatformProps, 'VerifyUser'>;
```

Comment:
What is the purpose of these? Isn't `AuthenticatorRouteComponentName = "ConfirmResetPassword" | "ConfirmSignIn" | "ConfirmSignUp" | "ConfirmVerifyUser" | "ForceNewPassword" | "ResetPassword" | "SetupTOTP" | "SignIn" | "SignUp" | "VerifyUser"` ? Not a fan of hardcoded strings.

Reply:
They are [function overloads](https://www.typescriptlang.org/docs/handbook/2/functions.html#function-overloads). In this case required for TS differentiate between the potential return types of `useAuthenticatorRoute` (I'm not huge of fan of this overload btw). Not certain i would call these _hard-coded strings_ since they are typed to `AuthenticatorRouteComponentName`?

Reply:
Sync-ed offline about removing the multiple overloads and using generics but that introduces other ts issues so won't block on this.